src/D2L.Bmx/Okta/Models/AuthenticateResponse.cs

```diff
@@ -21,11 +21,16 @@ internal record AuthenticateResponseEmbedded(
 	OktaMfaFactor[]? Factors
 );
 
+internal record OktaMfaQuestionProfile(
+	string QuestionText
+);
+
 internal record OktaMfaFactor(
 	string Id,
 	string FactorType,
 	string Provider,
-	string VendorName
+	string VendorName,
+	OktaMfaQuestionProfile Profile
```

Comment:
This I really do not like. 

It isn't straightforward to handle extracting nested value & do type discrimination based on FactorType for de serialization.

Without using a custom convertor at least AFAICS, so went with this approach.

Reply:
Isn't this what type discriminator is designed for? I don't see the need for a custom convertor.

Reply:
Couple of reasons,

1. Out of order discriminator isn't in as of .NET 8 ? https://github.com/dotnet/runtime/pull/97474
1. Need a bit more of a code change to differentiate between types without FactorType as a property (to not conflict with it's use as the type discriminator) 
1. It still looks kind of silly,  how even with type discrimination we would still need need to do some additional work to hoist QuestionText up from nested profile, ex:
      ```csharp
      // Binding to a nested property in polymorphic Profile
      [EditorBrowsable( EditorBrowsableState.Never )]
      public required OktaMfaQuestionProfile Profile { get; set; }
      public string QuestionText => Profile.QuestionText;
      ```

 All of which to me suggests, a custom convertor is the way to go, if we want to take it all the way (even if we discard the third point)

Reply:
1. We can use the STJ preview package.
2. Yes, which should be straightforward.
3. Not following... why?

Reply:
Moved to a separate PR targeting this ( #459 )since there is a few ways we can go about this.

re: 3,
Agreed it isn't necessary. Shouldn't really manipulate the Okta API types unnecessarily, after all its all still encapsulated within the context of the Okta authentication related code (no additional leak of implementation details involved).